src/main/java/io/cdap/plugin/AddFieldConfig.java

```diff
@@ -0,0 +1,148 @@
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import javax.annotation.Nullable;
+
+/**
+ * Config properties for the plugin.
+ */
+public class AddFieldConfig extends PluginConfig {
+  public static final String FIELD_NAME = "fieldName";
+  public static final String FIELD_VALUE = "fieldValue";
+  public static final String AS_UUID = "asUUID";
+
+  @Name(FIELD_NAME)
+  @Description("The name of the field to add. Must not already exist as an input field. The field type will be " +
+    "a nullable string.")
+  private String fieldName;
+
+  @Macro
+  @Nullable
+  @Name(FIELD_VALUE)
+  @Description("The value to set for the new field. If this is not specified, 'asUUID' must be set to true.")
+  private String fieldValue;
+
+  @Nullable
+  @Name(AS_UUID)
+  @Description("Generate a new UUID for the new field. If this is not true, 'fieldValue' must be specified.")
+  private Boolean asUUID;
+
+  public AddFieldConfig() {
+  }
+
+  public AddFieldConfig(String fieldName, @Nullable String fieldValue, @Nullable Boolean asUUID) {
+    this.fieldName = fieldName;
+    this.fieldValue = fieldValue;
+    this.asUUID = asUUID;
+  }
+
+  private AddFieldConfig(Builder builder) {
+    fieldName = builder.fieldName;
+    fieldValue = builder.fieldValue;
+    asUUID = builder.asUUID;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(AddFieldConfig copy) {
+    return builder()
+      .setFieldName(copy.fieldName)
+      .setFieldValue(copy.fieldValue)
+      .setAsUUID(copy.asUUID);
+  }
+
+  public String getFieldName() {
+    return fieldName;
+  }
+
+  @Nullable
+  public String getFieldValue() {
+    return fieldValue;
+  }
+
+  public Boolean getAsUUID() {
+    return asUUID == null ? false : asUUID;
+  }
+
+  public void validate(FailureCollector failureCollector, @Nullable Schema inputSchema) {
+    if (!containsMacro(FIELD_VALUE)) {
+      if (fieldValue == null && !getAsUUID()) {
+        failureCollector.addFailure("Must specify a field value or set 'asUUID' to true.", null)
+          .withConfigProperty(FIELD_VALUE)
+          .withConfigProperty(AS_UUID);
+      }
+
+      if (fieldValue != null && getAsUUID()) {
+        failureCollector.addFailure("Must not specify a field value or set 'asUUID' to false.", null)
+          .withConfigProperty(FIELD_VALUE)
+          .withConfigProperty(AS_UUID);
+      }
+    }
+
+    if (inputSchema != null) {
+      for (Schema.Field field : inputSchema.getFields()) {
+        if (field.getName().equals(fieldName)) {
+          failureCollector.addFailure(String.format("Field '%s' already exists in the input schema.", fieldName),
+                                      "Provide field that not present the input schema.")
+            .withConfigProperty(FIELD_NAME);
```

Comment:
you should also mark inputSchema field (there is a method to mark it invalid)

Reply:
Done.